### PR TITLE
Add French translation

### PR DESCRIPTION
**Pull request by lbourdois**
Hi @MKhalusova, 

In this PR you'll find the French translation of the course.
Only chapter 8 is missing. This is deliberate, as the deadline for certification is September 1, and the French translation will probably not be finished by then, so translating content that will no longer be viable when it's released doesn't seem relevant to me.

Indeed, most of the work is done, but in my experience, markdown tags or typos (spelling, keystrokes, etc.) have probably pop in. These are generally not obvious in the `mdx` files and require the `doc-builder`, which strangely doesn't work locally on my side, or which is accessible once have `write access`, which I don't yet have on this repo. 
Correcting them takes a few days and is fairly easy, but I'll be off from tomorrow evening for several weeks, so they probably won't arrive before September 1, although my aim is to be able to offer this complete translation by early September for the start of the university year.

Have a nice day,

Loïck

### CODE REVIEW

File: chapters/fr/chapter5/asr_models.mdx [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+dataset = load_dataset(
+    "facebook/multilingual_librispeech", "spanish", split="validation", streaming=True
+)
+sample = next(iter(dataset))
+```
+Encore une fois, nous allons inspecter la transcription du texte et écouter le segment audio:
+```python
+print(sample["text"])
+Audio(sample["audio"]["array"], rate=sample["audio"]["sampling_rate"])
+```**Sortie :**
```
Comment (MKhalusova) [inline chapters/fr/chapter5/asr_models.mdx:140]: The formatting here results in doc-builder errors.  A new line before `**Sortie :**` should fix it.

File: chapters/fr/chapter5/asr_models.mdx [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+```
+entonces te delelitarás en jehová y yo te haré subir sobre las alturas de la tierra y te daré á comer la heredad de jacob tu padre porque la boca de jehová lo ha hablado
+```
+C'est le texte cible que nous visons avec notre transcription Whisper. Bien que nous sachions maintenant que nous pouvons probablement faire mieux, puisque notre modèle va aussi prédire la ponctuation et la casse, qui ne sont pas présents dans la référence. Transmettons l'échantillon audio au pipeline pour obtenir notre prédiction de texte. Une chose à noter est que le pipeline _consomme_ le dictionnaire des entrées audio que nous entrons, ce qui signifie que le dictionnaire ne peut pas être réutilisé. Pour contourner ce problème, nous allons passer une _copie_ de l'échantillon audio, afin de pouvoir réutiliser le même échantillon audio dans les exemples de code suivants :
+```python
+pipe(sample["audio"].copy(), max_new_tokens=256, generate_kwargs={"task": "transcribe"})
+```**Sortie :**
```
Comment (MKhalusova) [inline chapters/fr/chapter5/asr_models.mdx:147]: Same formatting issue as above.

File: chapters/fr/chapter1/audio_data.mdx [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+La fréquence d'échantillonnage détermine l'intervalle de temps entre les échantillons audio successifs, ce qui a un impact sur la résolution temporelle des données audio. 
+Prenons un exemple : un son de 5 secondes à une fréquence d'échantillonnage de 16 000 Hz sera représenté comme une série de 80 000 valeurs, tandis que le même son de 5 secondes à une fréquence d'échantillonnage de 8 000 Hz sera représenté comme une série de 40 000 valeurs. 
+Les *transformers* qui résolvent les tâches audio traitent les exemples comme des séquences et s'appuient sur des mécanismes d'attention pour apprendre l'audio ou la représentation multimodale. 
+Étant donné que les séquences sont différentes pour les exemples audio à des taux d'échantillonnage différents, il sera difficile pour les modèles de généraliser entre les taux d'échantillonnage.
+**Le rééchantillonnage** est le processus de mise en correspondance des taux d'échantillonnage et fait partie du [prétraitement](preprocessing#resampling-the-audio-data) des données audio.
+
+## Amplitude et profondeur de bits
```
Comment (MKhalusova) [inline chapters/fr/chapter1/audio_data.mdx:20]: Please add a new line after the headers.

**Review by MKhalusova — CHANGES_REQUESTED** [review]
Thank you for taking over this massive effort and translating the whole course! 
Two more files need to be updated in order to build the course in a new language. `.github/workflows/build_documentation.yml` and `.github/workflows/build_pr_documentation.yml`.
Please add `fr` to the list of languages similar to how it's done [here](https://github.com/huggingface/audio-transformers-course/blob/360ba6cffda44d249029f790f03a93b08ec74cb8/.github/workflows/build_documentation.yml#L17).

Also, please review the markdown formatting. I spotted a couple of places where it breaks - see comments.

File: chapters/fr/chapter5/asr_models.mdx (hunk 1 above)
Comment (lbourdois) [reply chapters/fr/chapter5/asr_models.mdx]: Done

File: chapters/fr/chapter5/asr_models.mdx (hunk 2 above)
Comment (lbourdois) [reply chapters/fr/chapter5/asr_models.mdx]: Done

**Comment by lbourdois** [comment]
> Two more files need to be updated in order to build the course in a new language. .github/workflows/build_documentation.yml and .github/workflows/build_pr_documentation.yml.

Ah well, now I understand why I wasn't able to run the doc builder on my side. 🙃

I've corrected what you pointed out.

**Comment by MKhalusova** [comment]
The checks are failing ATM. Please run `make style` locally and try to build locally to identify the issues.

**Comment by lbourdois** [comment]
Classic NLP course mistake, shame on me for not thinking to do this line of code before opening my PR.
It should be fine now :)

**Comment by MKhalusova** [comment]
Almost there! The `audio_data.mdx` is missing a header, and that breaks the build.

**Comment by lbourdois** [comment]
I hope it will be good by now. If it isn't, I won't be able to make any changes until 3 weeks from now when I get back from vacation.

**Comment by MKhalusova** [comment]
Unfortunately, there's still an issue with building the docs. This time it appears to be formatting in `chapter3/quiz.mdx:135:174`. 
I would recommend to try building locally to catch this and other possible issues.

**Comment by lbourdois** [comment]
Having had quick access to the internet while on vacation, I tried to look at the problem quickly.
But not having my computer with me, I did it on my phone and on the browser.
After doing some crap, I simply deleted the quiz for the moment.
If it goes well, great, if not, I'll do a cleaner job after my vacation.

**Comment by lbourdois** [comment]
Hi @MKhalusova 
Indeed the `force-pushed main branch from 4d14d38 to ab97af9` replaced the `fr` with a `ru`. I re-added it.

File: chapters/fr/chapter6/tts_datasets.mdx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+Explorons quelques ensembles de données adaptés au TTS que vous pouvez trouver sur le  *Hub*.
+
+## LJSpeech
+
+[LJSpeech](https://huggingface.co/datasets/lj_speech) est un jeu de données composé de 13 100 échantillons audio de langue anglaise jumelés à leurs transcriptions correspondantes. Le jeu de données contient l'enregistrement d'un seul locuteur lisant des phrases de 7 livres en anglais. LJSpeech est souvent utilisé comme référence pour évaluer les modèles de TTS en raison de sa haute qualité audio et de son contenu linguistique diversifié.
+
+## Multilingual LibriSpeech
+[Multilingual LibriSpeech] (https://huggingface.co/datasets/facebook/multilingual_librispeech) est une extension multilingue du jeu de données LibriSpeech, qui est à la base une collection à grande échelle de livres audio lus en anglais. LibriSpeech multilingue inclus des langues supplémentaires, telles que l'allemand, le néerlandais, l'espagnol, le français, l'italien, le portugais et le polonais.
```
Comment (MKhalusova) [inline chapters/fr/chapter6/tts_datasets.mdx:17]:
```suggestion
[Multilingual LibriSpeech](https://huggingface.co/datasets/facebook/multilingual_librispeech) est une extension multilingue du jeu de données LibriSpeech, qui est à la base une collection à grande échelle de livres audio lus en anglais. LibriSpeech multilingue inclus des langues supplémentaires, telles que l'allemand, le néerlandais, l'espagnol, le français, l'italien, le portugais et le polonais. 
```

File: chapters/fr/chapter6/tts_datasets.mdx [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+Explorons quelques ensembles de données adaptés au TTS que vous pouvez trouver sur le  *Hub*.
+
+## LJSpeech
+
+[LJSpeech](https://huggingface.co/datasets/lj_speech) est un jeu de données composé de 13 100 échantillons audio de langue anglaise jumelés à leurs transcriptions correspondantes. Le jeu de données contient l'enregistrement d'un seul locuteur lisant des phrases de 7 livres en anglais. LJSpeech est souvent utilisé comme référence pour évaluer les modèles de TTS en raison de sa haute qualité audio et de son contenu linguistique diversifié.
+
+## Multilingual LibriSpeech
```
Comment (MKhalusova) [inline chapters/fr/chapter6/tts_datasets.mdx:16]:
```suggestion
## Multilingual LibriSpeech

```

File: chapters/fr/chapter6/tts_datasets.mdx [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+Il propose des enregistrements audio ainsi que des transcriptions alignées pour chaque langue. Le jeu de données fournit une ressource précieuse pour développer des systèmes et techniques TTS de synthèse vocale multilingue.
+
+## VCTK (*Voice Cloning Toolkit*)
+
+[VCTK](https://huggingface.co/datasets/vctk) est un jeu de données spécialement conçu pour la recherche et le développement de synthèse vocale.
+Il contient des enregistrements audio de 110 anglophones avec divers accents. Chaque orateur lit environ 400 phrases, qui ont été choisies dans un journal, et un paragraphe d'élicitation utilisé pour les archives de l'accent de la parole.
+C’est une ressource précieuse pour l’entraînement de modèles de TTS avec des voix et des accents variés, permettant une synthèse vocale plus naturelle et diversifiée.
+
+## Libri-TTS/ LibriTTS-R
```
Comment (MKhalusova) [inline chapters/fr/chapter6/tts_datasets.mdx:26]:
```suggestion
## Libri-TTS/ LibriTTS-R

```

File: chapters/fr/chapter6/pre-trained_models.mdx [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+
+Écoutons :
+
+```python
+from IPython.display import Audio
+
+Audio(speech, rate=16000)
+```
+
+Superbe ! Si vous souhaitez essayer MMS avec une autre langue, trouvez d'autres *checkpoints* `vits` appropriés sur le [*Hub*] (https://huggingface.co/models?filter=vits).
```
Comment (MKhalusova) [inline chapters/fr/chapter6/pre-trained_models.mdx:197]:
```suggestion
Superbe ! Si vous souhaitez essayer MMS avec une autre langue, trouvez d'autres *checkpoints* `vits` appropriés sur le [*Hub*](https://huggingface.co/models?filter=vits).
```

File: chapters/fr/chapter7/voice-assistant.mdx [hunk 8]
```diff
@@ -0,0 +1,361 @@
...
+
+Encore une fois, nous avons fait cela plusieurs fois maintenant, donc le processus sera très familier !
+
+## Détection des mots déclencheur
+
+La première étape du pipeline de l'assistant vocal consiste à détecter si le mot déclencheur a été prononcé, et nous devons donc trouver un modèle pré-entraîné approprié pour cette tâche ! Vous vous souviendrez de la section sur les [modèles pré-entraînés pour la classification audio](../chapiter4/classification_models) que [Speech Commands](https://huggingface.co/datasets/speech_commands) est un jeu de données de mots prononcés conçu pour évaluer les modèles de classification audio sur plus de 15 mots de commande simples comme `up`, `down`, `yes` et `no`, ainsi qu'un label `silence` pour classer l'absence de parole. Prenez une minute pour écouter les échantillons sur le visualisateur de jeux de données du *Hub* et vous familiariser à nouveau avec le jeu de données : [visualisateur] (https://huggingface.co/datasets/speech_commands/viewer/v0.01/train).
```
Comment (MKhalusova) [inline chapters/fr/chapter7/voice-assistant.mdx:37]:
```suggestion
La première étape du pipeline de l'assistant vocal consiste à détecter si le mot déclencheur a été prononcé, et nous devons donc trouver un modèle pré-entraîné approprié pour cette tâche ! Vous vous souviendrez de la section sur les [modèles pré-entraînés pour la classification audio](../chapiter4/classification_models) que [Speech Commands](https://huggingface.co/datasets/speech_commands) est un jeu de données de mots prononcés conçu pour évaluer les modèles de classification audio sur plus de 15 mots de commande simples comme `up`, `down`, `yes` et `no`, ainsi qu'un label `silence` pour classer l'absence de parole. Prenez une minute pour écouter les échantillons sur le visualisateur de jeux de données du *Hub* et vous familiariser à nouveau avec le jeu de données : [visualisateur](https://huggingface.co/datasets/speech_commands/viewer/v0.01/train).
```

File: chapters/fr/chapter7/voice-assistant.mdx [hunk 9]
```diff
@@ -0,0 +1,361 @@
...
+
+```
+'Hugging Face is a company that provides natural language processing and machine learning tools for developers. They'
+```
+
+Vous remarquerez à quel point l'inférence est rapide en utilisant l'API Inference. Nous n'avons qu'à envoyer un petit nombre de *tokens* de texte de notre machine locale au modèle hébergé, le coût de communication est donc très faible. Le LLM est hébergé sur des GPU, de sorte que l'inférence s'exécute très rapidement. Enfin, la réponse générée est transférée du modèle à notre machine locale, toujours avec un faible coût de communication.
+
+## Synthétiser la parole
+
+Et maintenant, nous sommes prêts à obtenir la sortie vocale finale ! Une fois de plus, nous utiliserons le modèle [SpeechT5 TTS] (https://huggingface.co/microsoft/speecht5_tts) de Microsoft mais vous pouvez utiliser n'importe quel modèle TTS de votre choix. Chargeons le processeur et le modèle :
```
Comment (MKhalusova) [inline chapters/fr/chapter7/voice-assistant.mdx:254]:
```suggestion
Et maintenant, nous sommes prêts à obtenir la sortie vocale finale ! Une fois de plus, nous utiliserons le modèle [SpeechT5 TTS](https://huggingface.co/microsoft/speecht5_tts) de Microsoft mais vous pouvez utiliser n'importe quel modèle TTS de votre choix. Chargeons le processeur et le modèle :
```

File: chapters/fr/chapter7/supplemental_reading.mdx [hunk 10]
```diff
@@ -0,0 +1,18 @@
+# Lectures et ressources complémentaires
+
+Cette unité a rassemblé de nombreux éléments des unités précédentes, en introduisant les tâches de traduction vocale (audio à audio), les assistants vocaux et séparation des locuteurs. 
+Le matériel de lecture supplémentaire est donc divisé en ces trois nouvelles tâches :
+
+Traduction vocale (audio à audio) :
+* [STST avec unités discrètes] (https://ai.facebook.com/blog/advancing-direct-speech-to-speech-modeling-with-discrete-units/) par Meta AI : une approche directe de la STST par le biais de modèles encodeur-décodeur.
```
Comment (MKhalusova) [inline chapters/fr/chapter7/supplemental_reading.mdx:7]:
```suggestion
* [STST avec unités discrètes](https://ai.facebook.com/blog/advancing-direct-speech-to-speech-modeling-with-discrete-units/) par Meta AI : une approche directe de la STST par le biais de modèles encodeur-décodeur.
```

**Review by MKhalusova — CHANGES_REQUESTED** [review]
We’re very close to merging this epic PR! Now that the pages are building, I can see that there are multiple places where there are issues with formatting, specifically with rendering links and code examples.

To fix these, please add:
- a new line before and after a code block 
- a new line after each header 
- a new line before and after a `<Tip>…</Tip>` block, a `<div>…</div>` block

I have also left a few formatting nits in the comments. Thanks for your work on this PR!

**Comment by lbourdois** [comment]
I've added a new line before and after a code block, after each header, before and after a <Tip>…</Tip> block,  and before a <div>…</div> block. And I've corrected the broken links.
I hope everything is ok.

**Comment by lbourdois** [comment]
Fingers crossed it's finally good

**Review by MKhalusova — APPROVED** [review]
Great work! 
We can merge now. Unit 8 (congratulations and how to get a certificate) is missing, but feel free to add it in a separate PR.

**Pull request merged**